autofill/autofill-impl/src/test/java/com/duckduckgo/autofill/jsbridge/request/AutofillJsonRequestParserTest.kt

```diff
@@ -30,43 +32,51 @@ class AutofillJsonRequestParserTest {
     @Test
     fun whenUsernameAndPasswordBothProvidedThenBothInResponse() = runTest {
         val parsed = "storeFormData_usernameAndPasswordProvided".parseStoreFormDataJson()
-        assertEquals("dax@duck.com", parsed.credentials.username)
-        assertEquals("123456", parsed.credentials.password)
+        assertEquals("dax@duck.com", parsed.credentials!!.username)
+        assertEquals("123456", parsed.credentials!!.password)
     }
 
     @Test
     fun whenUsernameAndPasswordBothMissingThenBothAreNull() = runTest {
         val parsed = "storeFormData_usernameAndPasswordMissing".parseStoreFormDataJson()
-        assertNull(parsed.credentials.username)
-        assertNull(parsed.credentials.password)
+        assertNull(parsed.credentials!!.username)
+        assertNull(parsed.credentials!!.password)
     }
 
     @Test
     fun whenUsernameAndPasswordBothNullThenBothAreNullInParsedObject() = runTest {
         val parsed = "storeFormData_usernameAndPasswordNull".parseStoreFormDataJson()
-        assertNull(parsed.credentials.username)
-        assertNull(parsed.credentials.password)
+        assertNull(parsed.credentials!!.username)
+        assertNull(parsed.credentials!!.password)
     }
 
     @Test
     fun whenAdditionalUnknownPropertiesInRequestThenStillParses() = runTest {
         val parsed = "storeFormData_additionalUnknownPropertiesIncluded".parseStoreFormDataJson()
-        assertEquals("dax@duck.com", parsed.credentials.username)
-        assertEquals("123456", parsed.credentials.password)
+        assertEquals("dax@duck.com", parsed.credentials!!.username)
+        assertEquals("123456", parsed.credentials!!.password)
     }
 
     @Test
     fun whenUsernameMissingThenPasswordPopulated() = runTest {
         val parsed = "storeFormData_usernameMissing".parseStoreFormDataJson()
-        assertNull(parsed.credentials.username)
-        assertEquals("123456", parsed.credentials.password)
+        assertNull(parsed.credentials!!.username)
+        assertEquals("123456", parsed.credentials!!.password)
     }
 
     @Test
     fun whenPasswordMissingThenUsernamePopulated() = runTest {
         val parsed = "storeFormData_passwordMissing".parseStoreFormDataJson()
-        assertEquals("dax@duck.com", parsed.credentials.username)
-        assertNull(parsed.credentials.password)
+        assertEquals("dax@duck.com", parsed.credentials!!.username)
+        assertNull(parsed.credentials!!.password)
+    }
+
+    @Test
+    fun whenTopLevelCredentialsObjectMissingThenParsesWithoutError() = runTest {
+        val parsed = "storeFormData_topLevelDataMissing".parseStoreFormDataJson()
+        assertNull(parsed.credentials)
+        assertNull(parsed.credentials!!.username)
```

Comment:
NIT: Seems redundant to check the username and password if credentialls is already null